src/schema/document/core.rs

```diff
@@ -0,0 +1,310 @@
+use std::collections::{BTreeMap, HashMap, HashSet};
+use std::net::Ipv6Addr;
```

Comment:
Not a fan of calling anything `core`, since it's typically not well defined what's core and what's not

In this case I'd rather move it to `mod.rs` or pick a more suitable name